commands/mr/mrutils/mrutils.go

```diff
@@ -215,15 +216,65 @@ func MRsFromArgs(f *cmdutils.Factory, args []string) ([]*gitlab.MergeRequest, gl
 
 }
 
-func GetOpenMRForBranch(apiClient *gitlab.Client, baseRepo glrepo.Interface, currentBranch string) (*gitlab.MergeRequest, error) {
+func GetOpenMRForBranch(apiClient *gitlab.Client, baseRepo glrepo.Interface, arg string) (*gitlab.MergeRequest, error) {
+	currentBranch := arg // Assume the user is using only 'branch', not 'OWNER:branch'
+	var owner string
+
+	// If the string contains a ':' then it is using the OWNER:branch format, split it and
+	// assign them to the appropriate values, do note that we do not expect multiple ':' as
+	// git does not allow ':' to be used on branch names
+	if strings.Contains(arg, ":") {
+		t := strings.Split(arg, ":")
+		owner = t[0]
+		currentBranch = t[1]
+	}
+
 	mrs, err := api.ListMRs(apiClient, baseRepo.FullName(), &gitlab.ListProjectMergeRequestsOptions{
 		SourceBranch: gitlab.String(currentBranch),
 		State:        gitlab.String("opened"),
 	})
-
 	if err != nil {
 		return nil, fmt.Errorf("failed to get open merge request for %q: %w", currentBranch, err)
 	}
+
+	// We got at least one match!
+	if len(mrs) > 0 {
+		// The user gave us an 'OWNER:' so try to match the merge request with it
+		if owner != "" {
+			for i := range mrs {
+				// We found a match!
+				if mrs[i].Author.Username == owner {
+					return mrs[i], nil
+				}
+			}
+			// No match, error out, tell the user which branch and which username we looked for
+			return nil, fmt.Errorf("no open merge request available for %q owned by @%s", currentBranch, owner)
+		}
+
+		// This is done after the 'OWNER:' check because we don't want to give the wrong MR
+		// to someone that **explicitly** asked for a OWNER.
+		if len(mrs) == 1 {
+			return mrs[0], nil
+		}
+
+		// No 'OWNER:' prompt the user to pick a merge request
+		mrMap := map[string]*gitlab.MergeRequest{}
+		var mrNames []string
+		for i := range mrs {
+			t := fmt.Sprintf("!%d (%s) by @%s", mrs[i].IID, currentBranch, mrs[i].Author.Username)
+			mrMap[t] = mrs[i]
+			mrNames = append(mrNames, t)
+		}
+		pickedMR := mrNames[0]
+		err := survey.AskOne(&survey.Select{
```

Comment:
I prefer you use our custom `prompt` package which wraps survey package. It helps us to stub responses in tests

```suggestion
		err := prompt.AskOne(&survey.Select{
```

Reply:
Fixed, if that is the case we might also want to fix it in `internal/glrepo/resolver.go` since I copied this `prompt user for input` part from there.

Reply:
Yeah I actually left it cos there was not test for that file